src/code_generators.cpp

```diff
@@ -122,8 +122,8 @@ std::string BaseGenerator::LastNamespacePart(const Namespace &ns) {
 std::string BaseGenerator::WrapInNameSpace(const Namespace *ns,
                                            const std::string &name) const {
   std::string qualified_name = qualifying_start_;
-  for (auto it = ns->components.begin(); it != ns->components.end(); ++it)
```

Comment:
it uses this old style syntax because we're still building for compilers as old as VS2010

Reply:
We're using c++11 features already. Does it even build on VS2010?

Reply:
Well, VS2010 is part of our CI, so yes (see AppVeyor results). And it supports about half of C++11 features :)

Reply:
That's a bit surprising, to be honest. Do you want me to revert those changes?

Reply:
I'm also for dropping support for old things, but perhaps in a different PR, after a public deprecation notice and sunset period.

Reply:
We either need to revert those changes or remove VS2010 from the CI. So for now maybe the former. We can restart the discussion on the latter elsewhere, if people feel strongly.